src/modules/configuration/src/lib/Configuration.c

```diff
@@ -0,0 +1,545 @@
+
+    int modelVersion = g_modelVersion;
+    int refreshInterval = g_refreshInterval;
+    int localManagementEnabled = g_localManagementEnabled ? 1 : 0;
+    int fullLoggingEnabled = g_fullLoggingEnabled ? 1 : 0;
+    int commandLoggingEnabled = g_commandLoggingEnabled ? 1 : 0;
+    int iotHubProtocol = g_iotHubProtocol;
+
+    JSON_Value* jsonValue = NULL;
+    JSON_Object* jsonObject = NULL;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (0 >= payloadSizeBytes))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet(%s, %s, %s, %d) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+
+    if ((MMI_OK == status) && (strcmp(componentName, g_configurationComponentName)))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+
+    if ((MMI_OK == status) && (strcmp(objectName, g_desiredConfigurationObject)))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet called for an unsupported object name (%s)", objectName);
+        status = EINVAL;
+    }
+
+    if (MMI_OK == status)
+    {
+        if (NULL != (payloadString = malloc(payloadSizeBytes + 1)))
+        {
+            memset(payloadString, 0, payloadSizeBytes + 1);
+            memcpy(payloadString, payload, payloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(ConfigurationGetLog(), "Failed to allocate %d bytes of memory, MmiSet failed", payloadSizeBytes + 1);
+            status = ENOMEM;
+        }
+    }
+
+    if (MMI_OK == status)
+    {
+        if (NULL == (jsonValue = json_parse_string(payloadString)))
+        {
+            OsConfigLogError(ConfigurationGetLog(), "json_parse_string(%s) failed, MmiSet failed", payloadString);
+            status = EINVAL;
+        }
+        else if (JSONObject != json_value_get_type(jsonValue))
+        {
+            OsConfigLogError(ConfigurationGetLog(), "json_value_get_type(%s) did not return JSONObject, MmiSet failed", payloadString);
+            status = EINVAL;
+        }
+        else if (NULL == (jsonObject = json_value_get_object(jsonValue)))
+        {
+            OsConfigLogError(ConfigurationGetLog(), "json_value_get_object(%s) failed, MmiSet failed", payloadString);
+            status = EINVAL;
+        }
+
+        if (MMI_OK == status)
+        {
+            if (0 != (modelVersion = (int)json_object_get_number(jsonObject, g_modelVersionObject)))
+            {
+                g_modelVersion = modelVersion;
+            }
+
+            if (0 != (refreshInterval = (int)json_object_get_number(jsonObject, g_refreshIntervalObject)))

```

Comment:
Should there be a check with `json_object_get_type()` before each `json_object_get_xyz()` ?